content/blog/2020/07/2020-07-03-github-checks-api-plugin-coding-phase-1.adoc

```diff
@@ -0,0 +1,109 @@
+Contributions for implementations on these platforms will be welcomed in the future.
+
+=== GitHub Checks API Implmentation
+
+Our work on supporting GitHub Checks API are mostly done by now, so after the releasing, Jenkins developers can then create GitHub checks for a GitHub branch source project by consuming our API.
+
+**Example:** To create a check run like:
+
+image:/images/post-images/2020/07-github-checks-api-plugin-coding-phase-1/created-check-run.png[Created Check Run]
+
+Consumers need to use our API in this way:
+
+[source, java]
+----
+
+ChecksDetails details = new ChecksDetailsBuilder()
+        .withName("Jenkins")
+        .withStatus(ChecksStatus.COMPLETED)
+        .withDetailsURL("https://ci.jenkins.io")
+        .withStartedAt(LocalDateTime.now(ZoneOffset.UTC))
+        .withCompletedAt(LocalDateTime.now(ZoneOffset.UTC))
+        .withConclusion(ChecksConclusion.SUCCESS)
+        .withOutput(new ChecksOutputBuilder()
+                .withTitle("Jenkins Check")
+                .withSummary("# A Successful Build")
+                .withText("## 0 Failures")
+                .withAnnotations(Arrays.asList(
+                        new ChecksAnnotationBuilder()
+                                .withPath("Jenkinsfile")
+                                .withLine(1)
+                                .withAnnotationLevel(ChecksAnnotationLevel.NOTICE)
+                                .withMessage("say hello to Jenkins")
+                                .withStartColumn(0)
+                                .withEndColumn(20)
+                                .withTitle("Hello Jenkins")
+                                .withRawDetails("a simple echo command")
+                                .build(),
+                        new ChecksAnnotationBuilder()
+                                .withPath("Jenkinsfile")
+                                .withLine(2)
+                                .withAnnotationLevel(ChecksAnnotationLevel.WARNING)
+                                .withMessage("say hello to GitHub Checks API")
+                                .withStartColumn(0)
+                                .withEndColumn(30)
+                                .withTitle("Hello GitHub Checks API")
+                                .withRawDetails("a simple echo command")
+                                .build()))
+                .build())
+        .withActions(Collections.singletonList(
+                new ChecksAction("formatting", "format code", "#0")))
+        .build();
+
+ChecksPublisher publisher = ChecksPublisherFactory.fromRun(run);
+publisher.publish(details);
+
+----
+
+== Future Works
+
+The next step is integrating our API into link:https://plugins.jenkins.io/warnings-ng/[Warnings Next Generation Plugin] and link:https://plugins.jenkins.io/code-coverage-api/[Code Coverage API Plugin] consume our API.
+Pipeline support is in our plan as well.
```

Comment:
We should make clear what Pipeline support means: that checks can be created and updated directly in a pipeline script (without requiring a consumer plugin that support the checks).